packages/trackerless-network/src/NetworkStack.ts

```diff
@@ -68,6 +71,34 @@ export class NetworkStack extends EventEmitter<NetworkStackEvents> {
         })
     }
 
+    async joinStreamPart(streamPartId: StreamPartID, neighborRequirement?: { minCount: number, timeout: number }): Promise<void> {
+        if (this.getStreamrNode().isProxiedStreamPart(streamPartId)) {
+            throw new Error(`Cannot join to ${streamPartId} as proxy connections have been set`)
+        }
+        await this.joinLayer0IfRequired(streamPartId)
+        setImmediate(async () => {
+            try {
+                await this.getStreamrNode().joinStream(streamPartId)
```

Comment:
If make the `createStream` method in `StreamrNode` a public method we can separate the state setting and joining logic. Then the `StreamrNode#joinStream` can only be called in cases where the stream object is newly created. Then `joinStream()` can always be called in parallel when necessary.

Reply:
Will be implemented in a separate PR